api/nameguard/nameguard.py

```diff
@@ -381,26 +390,45 @@ async def fake_eth_name_check(self, network_name, contract_address, token_id) ->
                 investigated_fields['.'.join(keys)] = name
             except KeyError:
                 pass
+
+        if contract_address in ens_contract_adresses:
+            if ALCHEMY_UNKNOWN_NAME.match(title):
+                unknown_name = f"[{res_json['id']['tokenId'][2:]}].eth"
+                investigated_fields['title'] = unknown_name
+        
+        title = investigated_fields['title']
+        del investigated_fields['title']
+        
+        return await self._fake_eth_name_check(network_name, contract_address, title, investigated_fields)
+        
+    async def _fake_eth_name_check(self, network_name, contract_address, title: str, fields: dict[str,str]) -> FakeEthNameCheckResult:
+        investigated_fields = {'title': title}
+        if 'title' in fields: del fields['title']
+        investigated_fields.update(fields)
+        
         
         if contract_address in ens_contract_adresses:
             if title == '':  # the name has never been registered
```

Comment:
This was Alchemy thing

Reply:
What should empty title (name) mean for ENS contracts? Now we return UNKNOWN_NFT.

Reply:
Ok.